Fix existing library folder path returning undefined

YouKnowBlom opened the pull request:
getDefaultPath in directorybrowser.js now returns a promise if options.path exists

**Issues**
#588

JustAMan commented on src/components/directorybrowser/directorybrowser.js:
```diff
@@ -223,7 +223,7 @@ define(['loading', 'dialogHelper', 'dom', 'listViewStyle', 'emby-input', 'paper-
-            Promise.resolve(options.path);
+            return Promise.resolve(options.path);
         } else {
             ApiClient.getJSON(ApiClient.getUrl("Environment/DefaultDirectoryBrowser")).then(
```
shouldn't this one be returned as well?

YouKnowBlom commented:
Good catch! I added a return to line 228 and tested without issue.

As a sidenote I should really work on my git skills 😄. I'm pretty sure I've reverted my mess up. Please do let me know if you notice anything I should've done differently

joshuaboniface commented:
Cherry-picked into 10.4.3.
